### ISSUE TITLE
Cooperative cancel for htlc receiver

### ISSUE THREAD

**Issue by kilrau** [ConnextProject/rest-api-client#39]
Currently the connext client doesn't allow a **receiver to cancel an incoming htlc** payment cooperatively with the sender.

This is a fairly important feature for us, since in the event of failures we rely on asking the receiver to cancel the htlc payment to cleanly cancel the unclaimed htlcs and cleanly return locked funds of a swap. This is currently not possibly with connext, but in xud the receiver failing to cancel the incoming payment (upon request of sender) results in locked up funds until htlc expiry, negative reputation events and eventually ban.

The other way around, htlc security relies on the **sender not being able to cancel an outgoing htlc**.

**TL;DR how it should work**: sender has to wait for htlc expiry, receiver can (cooperatively) cancel an unclaimed htlc any time
